docs/DEVELOPING.md

```diff
@@ -0,0 +1,192 @@
+# Developing LoopBack
+
+This document describes how to develop modules living in loopback-next monorepo.
+
+ - [Setting up development environment](#setting-up-development-environment)
+ - [Building the project](#building-the-project)
+ - [Running tests](#running-tests)
+ - [Coding rules](#coding-rules)
+ - [API documentation](#api-documentation)
+ - [Commit message guidelines](#commit-message-guidelines)
+ - [How to test infrastructure changes](#how-to-test-infrastructure-changes)
+
+## Setting up development environment
+
+Before you can start developing LoopBack, you need to install and configure few
+dependencies.
+
+ - [git](https://git-scm.com/): Github's [Set Up Git](https://help.github.com/articles/set-up-git/) guide is a good source of information.
+ - [Node.js 8.x (LTS)](https://nodejs.org/en/download/)
+
+You may want to configure your IDE or editor to get better support for
+TypeScript too.
+
+ - [VisualStudio Code](./VSCODE.md)
+ - _Missing your favorite IDE/editor here? Pull requests are welcome!_
+
+Before getting started, it is recommended to configure `git` so that it knows who you are:
+
+```sh
+$ git config --global user.name "J. Random User"
+$ git config --global user.email "j.random.user@example.com"
+```
+
+Please make sure this local email is also added to your [GitHub email list](https://github.com/settings/emails) so that your commits will be properly associated with your account and you will be promoted to Contributor once your first commit is landed.
+
+## Building the project
+
+Whenever you pull updates from GitHub or switch between feature branches,
+we recommend to do a full reinstall of all npm dependencies:
+
+```sh
+$ npm run clean:lerna && npm run bootstrap
```

Comment:
I think running just clean with bootstrap is fine. `npm run clean && npm run bootstrap`. Any reason to use `clean:lerna` here?

Reply:
`clean:lerna` removes `node_modules` files in all package folders (something like `rm -rf packages/*/node_modules`).

I vaguely remember having issues with dependency setup that were resolved by removing all node_modules folders.

If this is no longer a problem, then I am happy to change this line as you suggested.

@raymondfeng could you please chime in? AFAICT, `clean:lerna` was added by you in f1e59a2e2, which says:

> This change fixes the issue that `npm run release` fails if a new package is added to the monorepo before bootstrap is done. The new package doesn't have dependencies installed yet.

Can we safely assume that `clean:lerna` script is needed only for the release process?

Reply:
`clean:lerna` is mostly for `release` but it also be helpful in certain situations to force a fresh `bootstrap`.